xlml/apis/test_config.py

```diff
@@ -183,6 +185,22 @@ def setup_script(self) -> Optional[str]:
   def test_script(self) -> str:
     return '\n'.join(self.run_model_cmds)
 
+  @property
+  def startup_script(self) -> str:
+    if self.use_startup_script == False:
+      return None
+
+    main_command = '\n'.join(self.set_up_cmds + self.run_model_cmds)
+    final_command = f"""
```

Comment:
Can you explain a little bit about this `final_command`?

Reply:
Sure. This is a bit complicated since it is easy to add `startup script` to QR, but it is hard to check if workload finished and its exit status.

`main_command` contains the `setup command` and the `command to run workload`.

```
bash -c '{main_command} 2>&1 | tee /tmp/logs &
pid=$!
echo $pid > /tmp/main_process_id.txt
wait $pid
exit_status=$?
echo $exit_status > /tmp/process_exit_status.txt'
```

For final command, it will run `main_command` as a background process and save its process id into `/tmp/main_process_id.txt`. `/tmp/logs` will save all the logs from this workload. 
It will wait this process until it finish, which means the workload finished. 
It will save the process exit status(workload succeed/failed) into `/tmp/process_exit_status.txt`

In the `check_script` from `task.py`:
We will ssh into the first worker, keep checking if `/tmp/main_process_id.txt`'s process id finished. 
Then check the workload exit status from `/tmp/process_exit_status.txt` and printout the workload logs from `/tmp/logs`

Reply:
Added all bash scripts into `xlml/utils/startup_script_util.py`. It should be cleaner now!